src/main/scala/rocket/TLBPermissions.scala

```diff
@@ -18,6 +18,7 @@ case class TLBPermissions(
 object TLBPageLookup
 {
   private case class TLBFixedPermissions(
+    t: RegionType.T,
```

Comment:
I think you should have made this into 'e: Boolean = RegionType.PUT_EFFECTS, RegionType.GET_EFFECTS) contains x'. The way it is now it will not combine/optimize cases that actually end up with the same boolean value.